subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/DynamicVersionResolver.java

```diff
@@ -133,23 +134,25 @@ private RepositoryChainModuleResolution findLatestModule(LinkedList<RepositoryRe
                 failures.add(t);
                 continue;
             }
-            switch (request.resolveResult.getState()) {
+            switch (request.resolvedVersionMetadata.getState()) {
                 case Failed:
-                    failures.add(request.resolveResult.getFailure());
+                    failures.add(request.resolvedVersionMetadata.getFailure());
                     break;
                 case Missing:
                 case Unknown:
                     // Queue this up for checking again later
+                    // This is done because we're checking what we have locally in cache, and there may be nothing
```

Comment:
Thanks for clarifying that. 

:o: I think it would also make sense to clearly indicate the intention of a state by adding a comment e.g. `BuildableModuleComponentMetaDataResolveResult.State`, `BuildableModuleVersionListingResolveResult.State`.

Reply:
I'm not sure I follow you here. Can you explain what you have in mind?

Reply:
Pretty much this:

```
public interface BuildableModuleVersionListingResolveResult extends ResourceAwareResolveResult, ResolveResult {

    static enum State {
        Listed,      // indicates this, implies that
        Failed,      // indicates this, implies that
        Unknown // indicates this, implies that
    }

    ...
}
```

Reply:
Ah right, you meant on _existing_ states, not new ones :)